lib/CBattleCallback.cpp

```diff
@@ -879,37 +885,50 @@ TDmgRange CBattleInfoCallback::calculateDmgRange(const BattleAttackInfo &info) c
 
 	int attackDefenceDifference = 0;
 
-	double multAttackReduction = (100 - battleBonusValue (info.attackerBonuses, Selector::type(Bonus::GENERAL_ATTACK_REDUCTION))) / 100.0;
-	attackDefenceDifference += battleBonusValue (info.attackerBonuses, Selector::typeSubtype(Bonus::PRIMARY_SKILL, PrimarySkill::ATTACK)) * multAttackReduction;
-
-	double multDefenceReduction = (100 - battleBonusValue (info.attackerBonuses, Selector::type(Bonus::ENEMY_DEFENCE_REDUCTION))) / 100.0;
-	attackDefenceDifference -= info.defenderBonuses->Defense() * multDefenceReduction;
-
-	if(const Bonus *slayerEffect = info.attackerBonuses->getEffect(SpellID::SLAYER)) //slayer handling //TODO: apply only ONLY_MELEE_FIGHT / DISTANCE_FIGHT?
-	{
-		std::vector<int> affectedIds;
-		int spLevel = slayerEffect->val;
-
-		for(int g = 0; g < VLC->creh->creatures.size(); ++g)
+	if(attackerType->idNumber != CreatureID::ARROW_TOWERS)
+	{
+		double multAttackReduction =
+			(100 - battleBonusValue(info.attackerBonuses,
+			                        Selector::type(Bonus::GENERAL_ATTACK_REDUCTION)))
+			/ 100.0;
+		attackDefenceDifference +=
+			battleBonusValue(info.attackerBonuses,
+			                 Selector::typeSubtype(Bonus::PRIMARY_SKILL,
+			                                       PrimarySkill::ATTACK))
+			* multAttackReduction;

```

Comment:
It is standard to format code this way (align parameters by function boundary), but I don't insist - if you suggest another formatting style to fit into 80 - OK. Could you post an example?